src/core/stores/VNode.ts

```diff
@@ -91,18 +103,22 @@ export class VNode {
     }
     /**
      * Return the index of this VNode within its parent.
+     * Note: this method always takes account of range nodes.
```

Comment:
_children can be set, children can't. How would we handle all the splices etc on children?

Reply:
What about we don't and the guy just has to use the methods ? Maybe children might be an iterator to prevent accessing the array and be more efficient when walking the entire dom (as opposed to cloning/filtering the array like right now), then we would need an `hasChildren` method like the `hasChildNodes` of the dom. What do you think ? We would lose the ability to do `node.children.length`, but the API would be entirely consistent with respect to whether range nodes should be taken into account or not, we would not clone the children array anymore, and we would ensure people can't play with the children array itself directly.

Reply:
Agreed. But I meant the splices in the (private) methods of VNode itself.

Reply:
@dmo-odoo We need to look into this on Tuesday.